packages/domain/lib/src/use_cases/login_facebook.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:domain/domain.dart';
+import 'package:domain/src/use_cases/base/out_use_case.dart';
+
+class LoginFaceBookUseCase implements OutUseCase<Future<bool>> {
+  final AuthRepository authRepository;
+  final PreferencesLocalRepository preferences;
+
+  const LoginFaceBookUseCase({
+    required this.authRepository,
+    required this.preferences,
+  });
+
+  @override
+  Future<bool> call() async {
+    final UserEmailPass? user = await authRepository.loginWithFaceBook();
+    if (user == null) return false;
+    final List<UserEmailPass> users = await authRepository.fetchUsers();
+    final isAbleToLogin = users.any(
+      (element) =>
+          element.login == user.login && element.password == user.password,
+    );
+    isAbleToLogin ? await preferences.saveLoggedUser(user) : null;
+    return isAbleToLogin;
```

Comment:
searching the user by login and password should be done inside the query. so, in this case u can make authRepository.fetchUsers() return boolean

but it's great, that u did it not by usual loop but with any operator!